### PR TITLE
dasher: 2018-04-03 -> unstable-2021-04-25

### PR DESCRIPTION
**Pull request by moni-dz**
###### Motivation for this change
ZHF: #122042 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/accessibility/dasher/default.nix [hunk 1]
```diff
@@ -1,22 +1,35 @@
...
+, expat
+, itstool
+, gnome-doc-utils
+, which
+, at-spi2-core
+, dbus
+, libxslt
+, libxml2
+, speechSupport ? true
+, speechd ? null
```
Comment (SuperSandro2000) [inline pkgs/applications/accessibility/dasher/default.nix:18]:
```suggestion
, speechd
```

File: pkgs/applications/accessibility/dasher/default.nix [hunk 2]
```diff
@@ -1,22 +1,35 @@
...
 }:
 
 assert speechSupport -> speechd != null;
```
Comment (SuperSandro2000) [inline pkgs/applications/accessibility/dasher/default.nix:21]:
```suggestion
```

File: pkgs/applications/accessibility/dasher/default.nix [hunk 3]
```diff
@@ -1,22 +1,35 @@
...
 
-  src = fetchFromGitHub {
-    owner = "dasher-project";
-    repo = "dasher";
-    rev = "9ab12462e51d17a38c0ddc7f7ffe1cb5fe83b627";
-    sha256 = "1r9xn966nx3pv2bidd6i3pxmprvlw6insnsb38zabmac609h9d9s";
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "GNOME";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/applications/accessibility/dasher/default.nix:30]:
```suggestion
    repo = "dasher";;
```

File: pkgs/applications/accessibility/dasher/default.nix [hunk 4]
```diff
@@ -1,22 +1,35 @@
...
+, libxslt
+, libxml2
+, speechSupport ? true
+, speechd ? null
 }:
 
 assert speechSupport -> speechd != null;
 
-stdenv.mkDerivation {
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/accessibility/dasher/default.nix:23]:
```suggestion
stdenv.mkDerivation {
```

File: pkgs/applications/accessibility/dasher/default.nix [hunk 5]
```diff
@@ -33,23 +46,26 @@ stdenv.mkDerivation {
...
+    at-spi2-core
+    dbus
   ] ++ lib.optional speechSupport speechd;
 
   meta = {
     homepage = "http://www.inference.org.uk/dasher/";
     description = "Information-efficient text-entry interface, driven by natural continuous pointing gestures";
     license = lib.licenses.gpl2;
     maintainers = [ lib.maintainers.Profpatsch ];
+    mainProgram = pname;
```
Comment (SuperSandro2000) [inline pkgs/applications/accessibility/dasher/default.nix:68]:
```suggestion
```

mainProgram defaults to pname. Please do not add the default explicitly.

File: pkgs/applications/accessibility/dasher/default.nix [hunk 6]
```diff
@@ -33,23 +46,26 @@ stdenv.mkDerivation {
...
     expat
+    itstool
     # at-spi2 needs dbus to be recognized by pkg-config
-    at-spi2-core dbus
+    at-spi2-core
+    dbus
   ] ++ lib.optional speechSupport speechd;
 
   meta = {
     homepage = "http://www.inference.org.uk/dasher/";
```
Comment (SuperSandro2000) [inline pkgs/applications/accessibility/dasher/default.nix:64]:
```suggestion
    homepage = "https://www.inference.org.uk/dasher/";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by fortuneteller2k** [comment]
Result of `nixpkgs-review pr 122470` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dasher</li>
  </ul>
</details>

**Pull request merged**